noir-r1cs/src/noir_proof_scheme.rs

```diff
@@ -16,8 +20,9 @@ use {
 };
 
 /// A scheme for proving a Noir program.
-#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct NoirProofScheme {
+    pub program:           Program<NoirFieldElement>,
     pub r1cs:              R1CS,
     pub witness_builders:  Vec<WitnessBuilder>,
     pub witness_generator: NoirWitnessGenerator,
```

Comment:
`witness_builders`, `witness_generator` and now also a `generate_witness` fn entirely separate from these? This is needs to be cleaned up soon, but maybe not in this PR.